pyOCD/board/mbed_board.py

```diff
@@ -150,7 +150,8 @@ def listConnectedBoards(dap_class=DAPAccess):
 
     @staticmethod
     def getAllConnectedBoards(dap_class=DAPAccess, close=False, blocking=True,
-                              target_override=None, frequency=1000000):
+                              target_override=None, frequency=1000000,
+                              board_id = None):
```

Comment:
I can do this -- however, this would imply we are willing to ignore the (remote but possible) case of multiple devices with the same serial number being connected.

It's that an acceptable limitation?